lib/util/merge-confidence/index.spec.ts

```diff
@@ -43,157 +49,329 @@ describe('util/merge-confidence/index', () => {
     });
   });
 
-  describe('getMergeConfidenceLevel()', () => {
+  describe('API calling functions', () => {
+    const envOrg: NodeJS.ProcessEnv = process.env;
+    const hostRule: HostRule = {
+      hostType: 'merge-confidence',
+      token: 'some-token',
+    };
+
     beforeEach(() => {
-      hostRules.clear();
+      jest.resetAllMocks();
+      process.env = {
+        ...envOrg,
+        RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL: apiBaseUrl,
+      };
```

Comment:
@viceice I tried my best but couldn't refactor this without tests failing when executed together.

i.e. each test would pass individually but the last two would fail if the whole file was run in one go

Reply:
whats the error?

Reply:
It looks like it's hitting a http mock cache or something like that as it doesn't even reach the mocks:

```
    Missing mocks!
     *   GET https://www.baseurl.com/api/mc/availability
```